ledger/internal/apptxn_test.go

```diff
@@ -3058,3 +3058,222 @@ check:
 	txns(t, l, eval, &fundA, &callA)
 	endBlock(t, l, eval)
 }
+
+func TestForeignAppAccountsAccessible(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	genBalances, addrs, _ := ledgertesting.NewTestGenesis()
+	testConsensusRange(t, 32, 0, func(t *testing.T, ver int) {
+		dl := NewDoubleLedger(t, genBalances, consensusByNumber[ver])
+		defer dl.Close()
+
+		appA := txntest.Txn{
+			Type:   "appl",
+			Sender: addrs[0],
+			ApprovalProgram: main(`
+int 3
+int 3
+==
+assert
+`),
```

Comment:
You can actually leave this out and txntest.Txn will supply a passing ApprovalProgram.